frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx

```diff
@@ -41,13 +43,17 @@ const ChartSettingFieldPicker = ({
       {columnKey && (
         <SettingsIcon
           name="ellipsis"
+          ref={settingsRef}
           onClick={() => {
-            onShowWidget({
-              id: "column_settings",
-              props: {
-                initialKey: columnKey,
+            onShowWidget(
+              {
+                id: "column_settings",
+                props: {
+                  initialKey: columnKey,
+                },
               },
-            });
+              settingsRef.current,
```

Comment:
I think it might be possible to use a target from an event from `onClick`. This would help simplify the code quite a bit by being able to remove `useRef` and keep the code relatively the same.

Reply:
+1

Reply:
Ohhh, yeah, ill give that a shot